forum/qa-plugin/report-reason/script.js

```diff
@@ -0,0 +1,116 @@
+document.addEventListener("DOMContentLoaded", () => {
+    
+    const flagboxPopup = document.querySelector("#flagbox-popup");
+    const wrap = document.querySelector(".qa-flag-reasons-wrap");
+    const body = document.body;
+    const closer = document.querySelector(".close-preview-btn");
+    const sendButton = document.querySelector(".qa-go-flag-send-button");
+    const errorMessage = "Błąd serwera. Proszę spróbować za jakiś czas";
+    const errorPopup = document.querySelector("#qa-spam-reason-error");
+    const tooManyReportError = "Zbyt dużo zgłoszeń z tego adresu IP. Spróbuj za godzinę";
+    const reportReasonEmptyError = "Wybierz powód zgłoszenia!";
+    const commentSubmitButton = document.querySelector('.qa-form-tall-button-comment')
+
+    function showPopup() {
+        flagboxPopup.classList.remove("hide");
+        flagboxPopup.classList.add("show");
+    }
+    function hidePopup() {
+        flagboxPopup.classList.remove("show");
+        flagboxPopup.classList.add("hide");
+    }
+    function removeAllChildFromErrorPopup() {
+        while (errorPopup.firstChild) {
+             errorPopup.removeChild(errorPopup.firstChild);
+        }
+    }
+    function showError(errorText) {
+        const paragraph = document.createElement("p");
+        paragraph.textContent = errorText;
+                    
+        removeAllChildFromErrorPopup();
+        errorPopup.appendChild(paragraph);
```

Comment:
Dodanie nowego elementu do DOM lepiej robić po wszystkich jego manipulacjach w danym czasie - przesuń to linię niżej.